pkg/controller/trainjob_controller.go

```diff
@@ -143,6 +176,34 @@ func (r *TrainJobReconciler) reconcileObjects(ctx context.Context, runtime jobru
 	return creationSucceeded, nil
 }
 
+func (r *TrainJobReconciler) Create(e event.TypedCreateEvent[*trainer.TrainJob]) bool {
+	r.log.WithValues("trainJob", klog.KObj(e.Object)).Info("TrainJob create event")
+	defer r.notifyWatchers(nil, e.Object)
+	return true
+}
+
+func (r *TrainJobReconciler) Delete(e event.TypedDeleteEvent[*trainer.TrainJob]) bool {
+	r.log.WithValues("trainJob", klog.KObj(e.Object)).Info("TrainJob delete event")
+	defer r.notifyWatchers(e.Object, nil)
+	return true
+}
+
+func (r *TrainJobReconciler) Update(e event.TypedUpdateEvent[*trainer.TrainJob]) bool {
+	r.log.WithValues("trainJob", klog.KObj(e.ObjectNew)).Info("TrainJob update event")
```

Comment:
Just to confirm, it's not necessary to dispatch updates? Is the runtime reference immutable?

Reply:
Yes, that will be immutable in https://github.com/kubeflow/trainer/pull/2619
However, the above implementation is not finalized. So, let's dispatch updates just for 2.0.
After we complete the above PR, we can remove the Update dispatcher.

Reply:
Done.